src/qengine/opencl.cpp

```diff
@@ -195,6 +194,10 @@ cl::Event QEngineOCL::QueueCall(
         ocl.call.setArg(args.size(), cl::Local(localBuffSize));
     }
 
+#if ENABLE_RASPBERRYPI
```

Comment:
doe3000, thanks for checking up on our implementation! Yes, I will file an issue, because I think this is a bug.

That `clFinish()` is actually a method of our `QEngineOCL` type. All it does is wait on a list of events; it's not a hard queue finish. The same exact list of events is passed as the wait list to the kernel, otherwise. So, there should be no logical difference between passing the wait list to the kernel and finishing it before calling the kernel, except that the former could continue asynchronously.

I just figured, we're asking a _lot_ of the Raspberry Pi's limited resources, to have this asynchronous behavior. But I will file an issue.